pkg/collector/corechecks/network-devices/cisco-sdwan/client/types.go

```diff
@@ -0,0 +1,388 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package client
+
+type viewKeys struct {
+	UniqueKey     []string `json:"uniqueKey"`
+	PreferenceKey string   `json:"preferenceKey"`
+}
+
+type fields struct {
+	Property string `json:"property"`
+	DataType string `json:"dataType"`
+}
+
+type columns struct {
+	Title         string `json:"title"`
+	Property      string `json:"property"`
+	DataType      string `json:"dataType"`
+	DisplayFormat string `json:"displayFormat,omitempty"`
+	InputFormat   string `json:"inputFormat,omitempty"`
+}
+
+type chart struct {
+}
+
+type header struct {
+	GeneratedOn int64     `json:"generatedOn"`
+	ViewKeys    viewKeys  `json:"viewKeys"`
+	Columns     []columns `json:"columns"`
+	Fields      []fields  `json:"fields"`
+	Chart       chart     `json:"chart"`
+}
+
+// PageInfo contains API pagination
+type PageInfo struct {
+	// Index based pagination
+	StartID     string `json:"startId"`
+	EndID       string `json:"endId"`
+	MoreEntries bool   `json:"moreEntries"`
+	Count       int    `json:"count"`
+
+	// ScrollId based pagination
+	ScrollID    string `json:"scrollId"`
+	HasMoreData bool   `json:"hasMoreData"`
+}
+
+// Content is all supported data of this client
+type Content interface {
+	DeviceStatistics |
+		Device |
+		InterfaceStats |
+		DeviceCounters |
+		InterfaceState |
+		CEdgeInterfaceState |
+		AppRouteStatistics |
+		ControlConnections |
+		OMPPeer |
+		BFDSession
+}
+
+// Response is a generic stuct for API responses
+type Response[T Content] struct {
+	Header   header   `json:"header"`
+	Data     []T      `json:"data"`
+	PageInfo PageInfo `json:"pageInfo"`
+}
+
+// Device /dataservice/device
+type Device struct {
+	DeviceID            string   `json:"deviceId"`
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ Types defined here will unmarshal all fields received in the response payload even if they are not used by the Cisco SD-WAN corecheck. 
It has the benefits of autocompletion when adding new features to the corecheck, but it will also mean we have more chances to fail unmarshalling (e.g. If a field changes type on the API side from `int` to `float`, we will fail unmarshalling the payload even if we don't use the field)

Should we only unmarshal the fields we use in the corecheck?